Add CodeQL

jkmassel opened the pull request:
This PR adds `CodeQL` to see if it can surface issues for us. We get it for free, so might as well.

jkmassel committed:
Add CodeQL

oguzkocer commented on .github/workflows/codeql.yml:
```diff
@@ -0,0 +1,93 @@
+      run: |
+        make setup-rust-android-targets
+        cd native/kotlin
+        ./gradlew cargoBuild --info
```
What's the intention here? `./gradlew cargoBuild` is basically the same as running `cargo build --release -p wp_api`. (not exactly, but very minor differences) So, it doesn't really build anything Kotlin related.

You probably want something like `./gradlew :api:android assemble` instead, but hard to say without knowing the intention.

jkmassel replied:
You're right about wanting `./gradlew :api:android assemble`, but I'm getting this error:

> Cannot locate tasks that match ':api:android' as task 'android' not found in project ':api'.

Am I holding it wrong?

oguzkocer replied:
Oh, sorry, it looks like I put a space where I should have put a `:` 🤦 It should be `./gradlew :api:android:assemble`. Sorry about that!

Having said that, maybe do `./gradlew :api:android:assembleRelease` instead. We don't have any debug/release specific config, and running `assemble` would do both. So, `assembleRelease` should be faster without any downside.

jkmassel committed:
Build for Android

As noted in https://github.com/Automattic/wordpress-rs/pull/251#discussion_r1712862662
